src/util/privkey.rs

```diff
@@ -110,13 +109,18 @@ impl Display for Privkey {
         };
         fmt.write_str(&privkey)
     }
-}
 
-impl FromStr for Privkey {
-    type Err = encode::Error;
+    /// Get WIF encoding of this private key.
+    #[inline]
+    pub fn to_wif(&self) -> Result<String, fmt::Error> {
```

Comment:
How can `to_wif` ever fail? Shouldn't the result of `fmt_wif` be unwrapped in this function and the return type be `String`